ffi_internals/src/alias_resolution.rs

```diff
@@ -160,54 +195,71 @@ pub(super) fn resolve_type_alias(type_name: &Ident, relevant_modules: &[String])
+/// constructed from its other data. The resulting `Item` can be safely sent back to the
+/// `TokenStream` and will no longer have the attribute used by the `alias_resolution` attribute
+/// macro.
 ///
-fn update_alias_map(resolution_key: String, new_aliases: HashMap<String, AliasDefinition>) {
+fn strip_alias_attribute(item_type: &ItemType) -> Item {
+    let stripped_attrs: Vec<Attribute> = item_type
+        .attrs
+        .clone()
+        .into_iter()
+        .filter(|a| parse_nested_alias_meta(a).is_none())
+        .collect();
+    let new_item_type = ItemType {
+        attrs: stripped_attrs,
+        vis: item_type.vis.clone(),
+        type_token: item_type.type_token,
+        ident: item_type.ident.clone(),
+        generics: item_type.generics.clone(),
+        eq_token: item_type.eq_token,
+        ty: item_type.ty.clone(),
+        semi_token: item_type.semi_token,
+    };
+    Item::Type(new_item_type)
+}
+
+/// Updates the `alias_map` file on disk with a new map of aliases under the `resolution_key`.
+///
+/// # Errors
+///
+/// This function will return an error if anything goes wrong when getting a lock on the file path,
+/// reading or writing the file, or parsing the file's JSON.
+///
+fn update_alias_map(
+    resolution_key: String,
+    new_aliases: HashMap<String, AliasDefinition>,
+) -> Result<(), Error> {
     // Read the existing file so we can add to it, or, if it doesn't exist, initialize an empty
     // `HashMap`.
-    let alias_map_path = ALIAS_MAP_PATH.lock().unwrap();
+    let alias_map_path = ALIAS_MAP_PATH.lock()?;
     let mut map: HashMap<String, HashMap<String, AliasDefinition>> =
         match std::fs::OpenOptions::new()
             .read(true)
             .open(&*alias_map_path)
         {
             Ok(file) => {
-                let mut reader = std::io::BufReader::new(file);
-                if reader.fill_buf().ok().unwrap().is_empty() {
-                    HashMap::new()
-                } else {
-                    match serde_json::from_reader(reader) {
-                        Ok(result) => result,
-                        Err(e) => panic!("Can't parse the file {}: {}", alias_map_path, e),
-                    }
-                }
+                let reader = std::io::BufReader::new(file);
+                serde_json::from_reader(reader)?
             }
             Err(_) => HashMap::new(),
         };
-    map.insert(resolution_key, new_aliases);
 
-    // Write `map`, which now also inclues the new alias resolution data for `module`, back to disk.
-    let file = std::fs::OpenOptions::new()
-        .write(true)
-        .create(true)
-        .open(&*alias_map_path)
-        .unwrap_or_else(|e| panic!("Error opening file to write {}: {}", alias_map_path, e));
+    *map.entry(resolution_key).or_default() = new_aliases;
 
-    serde_json::to_writer(file, &map)
-        .unwrap_or_else(|e| println!("Error writing file {}: {}", alias_map_path, e));
```

Comment:
I think this was the actual problem. `to_writer` writes it to whatever IOStream you give it, but I don't see anything in its docs about *completely* overwriting the contents if it happens to be a file. [`std::fs::write`](https://doc.rust-lang.org/std/fs/fn.write.html) replaces the entire contents of the file, so we won't have partially overwritten (and therefore borked JSON) files anymore.